LoopKit/UserAlert.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  UserAlert.swift
+//  LoopKit
+//
+//  Created by Rick Pasetto on 4/8/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+/// Protocol that describes any class that handles Alert posting.
+public protocol UserAlertHandler: class {
+    /// Schedule the given alert for posting.
+    func scheduleAlert(_ alert: UserAlert)
+    /// Unschedule any pending alerts with the given identifier.
+    func unscheduleAlert(managerIdentifier: String, typeIdentifier: UserAlert.TypeIdentifier)
```

Comment:
managerIdentifier does not need to be part of the protocol, it can be fetched from the manager itself, if the implementing object needs it (it may not).

Reply:
I think implementing these methods on the DeviceManagerDelegate as I commented above would make this relationship more natural.

Reply:
That's the reason I had to put it here: the implementing object doesn't need it.  I considered making a tiny `UserAlert.Identifier` struct with both of these in it to use instead but I thought it would have been overkill.   But it would have registered my intent better:  Ideally, you'd construct a `UserAlert.Identifier` at the source (with both `managerIdentifier` and `typeIdentifier`), but the "destination" user could only get out an `identifier` (of type `TypeIdentifier`) because the composition is opaque.  I liked this design better, but, again, I was thinking it might be overkill.